lib/remap-functions/src/includes.rs

```diff
@@ -0,0 +1,166 @@
+use remap::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct Includes;
+
+impl Function for Includes {
+    fn identifier(&self) -> &'static str {
+        "includes"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                accepts: |v| matches!(v, Value::Array(_)),
+                required: true,
+            },
+            Parameter {
+                keyword: "item",
+                accepts: |_| true,
+                required: true,
+            },
+        ]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
+        let value = arguments.required("value")?.boxed();
+        let item = arguments.required("item")?.boxed();
+
+        Ok(Box::new(IncludesFn { value, item }))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct IncludesFn {
+    value: Box<dyn Expression>,
+    item: Box<dyn Expression>,
+}
+
+impl Expression for IncludesFn {
+    fn execute(&self, state: &mut state::Program, object: &mut dyn Object) -> Result<Value> {
+        let list = self.value.execute(state, object)?.try_array()?;
+
+        let included = if list.is_empty() {
+            false
+        } else {
+            let item = self.item.execute(state, object)?;
```

Comment:
Do we need to document that `item` is only evaluated if the list is not empty?

Reply:
Personally, I don't believe so. Getting `false` with an empty array as input would be the intuitively expected behavior, so I don't think we need to expose those internals. Could be persuaded otherwise, though.

Reply:
I mean, from what I can tell, `item` is not *executed* if the list is empty. So, if `item` is an expression with side-effects (think `log` or something else), it won't happen for empty lists.

Realistically this kind of usage is somewhere between rare and bonkers with our current set of remap functions, so it still might not be an issue, but it could be a surprise at some point.

Reply:
Oh wait, sorry, you mean as a code comment. Yeah, would be good to make a note.

Reply:
I meant as something writers of scripts should be aware of.

I thought of a more realistic example: `if includes(.array, del(.some.other)) …`
In this case, `del` will only happen if `.array` has some element. That might be surprising to script writers, and should either be documented or the code reordered to always execute `item`.

Reply:
This is worth keeping in mind for *all* functions. A good rule of thumb is that we shouldn't have to explain a function's behavior in this way. It should flow naturally from the way we've structured the language. Returning a quick `false` when the array is empty, as I do here, violates one of those tenets and thus should be changed. Excellent catch!